src/panels/KaitoManagePanel.ts

```diff
@@ -0,0 +1,207 @@
+    ) {
+        this.clusterName = clusterName;
+        this.subscriptionId = subscriptionId;
+        this.resourceGroupName = resourceGroupName;
+        this.armId = armId;
+        this.kubectl = kubectl;
+        this.kubeConfigFilePath = kubeConfigFilePath;
+        this.models = models;
+    }
+
+    getTitle(): string {
+        return `Manage Kaito Models`;
+    }
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            models: this.models,
+        };
+    }
+    getTelemetryDefinition(): TelemetryDefinition<"kaitoManage"> {
+        return {
+            monitorUpdateRequest: false,
+            deleteWorkspaceRequest: true,
+            redeployWorkspaceRequest: true,
+            getLogsRequest: true,
+        };
+    }
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            monitorUpdateRequest: () => {
+                this.handleMonitorUpdateRequest(webview);
+            },
+            deleteWorkspaceRequest: (params) => {
+                this.handleDeleteWorkspaceRequest(params.model, webview);
+            },
+            redeployWorkspaceRequest: (params) => {
+                this.handleRedeployWorkspaceRequest(params.modelName, params.modelYaml, webview);
+            },
+            getLogsRequest: () => {
+                this.handleGetLogsRequest();
+            },
+        };
+    }
+    // State tracker for ongoing operations
+    private operatingState: Record<string, boolean> = {};
+    private async handleDeleteWorkspaceRequest(model: string, webview: MessageSink<ToWebViewMsgDef>) {
+        if (this.operatingState[model]) {
+            vscode.window.showErrorMessage(`Operation in progress for 'workspace-${model}'. Please wait.`);
+            return;
+        }
+        this.operatingState[model] = true;
+        try {
+            await longRunning(`Deleting 'workspace-${model}'`, async () => {
+                const command = `delete workspace workspace-${model}`;
+                const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+                if (failed(kubectlresult)) {
+                    vscode.window.showErrorMessage(
+                        `There was an error deleting 'workspace-${model}'. ${kubectlresult.error}`,
+                    );
+                    return;
+                }
+            });
+            vscode.window.showInformationMessage(`'workspace-${model}' was deleted successfully`);
+            await this.updateModels(webview);
+        } finally {
+            this.operatingState[model] = false;
+        }
+    }
+
+    // Deletes the workspace and redeploys it
+    private async handleRedeployWorkspaceRequest(
+        modelName: string,
+        modelYaml: string,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        if (this.operatingState[modelName]) {
```

Comment:
💡 what is the expected value of this pleas? I guess `boolean` (But worth add context as comment)

Reply:
Would be boolean, will add comments 👍